activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -124,10 +124,23 @@ def missing(*associations)
         associations.each do |association|
           reflection = scope_association_reflection(association)
           @scope.left_outer_joins!(association)
+          association_primary_key = reflection.association_primary_key
           if reflection.options[:class_name]
-            @scope.where!(association => { reflection.association_primary_key => nil })
+            if association_primary_key.is_a?(Array)
+              association_primary_key.each do |key|
+                @scope.where!(association => { key => nil })
+              end
+            else
+              @scope.where!(association => { reflection.association_primary_key => nil })
+            end
```

Comment:
I think, you can replace the special handling by something like:
```ruby
association_primary_key = Array(reflection.association_primary_key)

@scope.where!(association => { association_primary_key => [nil] * association_primary_key.size })
```

Reply:
This won't work as the array will be checked multiple times instead of checking the individual values each respectively; resulting in a query like: 

```
WHERE "comments"."[""post_id"", ""account_id""]" IS NOT NULL AND "comments"."[""post_id"", ""account_id""]" IS NOT NULL
```

If you have any suggestions on how to overcome that, it would be welcome, as it would clean up this piece quite a bit.

Reply:
Try this?
```ruby
association_primary_key = Array(reflection.association_primary_key)

@scope.where!(association => association_primary_key.to_h { |column| [column, nil] })
```

Reply:
Tweaked that a little and it cleaned up nice, thank you! 

Still not 100% happy with the name `association_primary_key_hash` 🤔